framework/src/main/java/org/checkerframework/framework/type/AsSuperVisitor.java

```diff
@@ -368,7 +368,13 @@ public AnnotatedTypeMirror visitDeclared_Primitive(
     @Override
     public AnnotatedTypeMirror visitDeclared_Typevar(
             AnnotatedDeclaredType type, AnnotatedTypeVariable superType, Void p) {
-        AnnotatedTypeMirror upperBound = visit(type, superType.getUpperBound(), p).asUse();
+        // setUpperBound() may have a side effect on parameter "type" when the upper bound of
+        // "superType" equals to "type" (referencing the same object: changes will be shared)
+        // copy before visiting to avoid
+        // without fix, this would fail:
+        // https://github.com/typetools/checker-framework/blob/ed340b2dfa1e51bbc0a7313f22638179d15bf2df/checker/tests/nullness/Issue2432b.java
```

Comment:
Do you want to extract this fix and test case into a separate PR?

Reply:
The error depends on the changes in this PR to reproduce. It would be hard to extract based on master.